File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

string Grade_Level(int number);
int main(void)
{
    //Initialization
    int letters = 0;
    int words = 1;
    int sentences = 0;
    //Get User Input
    string Text = get_string("Text: ");
    //length of User Input
    int len = strlen(Text);
    //Counting letters, words and sentences.
    for (int i = 0; i < len ; i++)
    {
        //Count letters
        if (isalpha(Text[i]))
        {
            letters++;
        }
        else if (isspace(Text[i]))
        {
            words++;
        }
        else if (Text[i] == '.' || Text[i] == '?' || Text[i] == '!')
        {
            sentences++;
        }
    }
    //Average Letters per 100 words
    float L =  100 * ((float) letters / (float) words) ;
    //Average Sentences per 100 words
    float S = 100 * ((float) sentences / (float) words);
    //Using Coleman-Liau index
    float index =  0.0588 * L - 0.296 * S - 15.8;
    //Print appropriate Grade Level
    int Count = round(index);
    if (Count < 1 || Count >= 16)
    {
         string Grade = Grade_Level(Count);
         printf("%s\n", Grade);
    }
    else
    {
        printf("Grade %i\n", Count);
    }
}

string Grade_Level(int number)
{
    string Outputs = "";
    if (number < 1)
    {
        Outputs = "Before Grade 1";
    }
    else if (number >= 16)
    {
        Outputs = "Grade 16+";
    }
    return Outputs;
}

File: readability.py
from cs50 import get_string


def main():
    # Initialization
    letters = 0
    words = 1
    sentences = 0
    # Get User Input
    text = get_string("text: ")
    # length of User Input
    length = len(text)
    # Counting letters, words and sentences.
    for i in range(length):
        # Count letters
        if (text[i].isalpha()):
            letters += 1
        elif (text[i].isspace()):
            words += 1
        elif (text[i] == '.' or text[i] == '?' or text[i] == '!'):
            sentences += 1
    # Average Letters per 100 words
    L = float(100 * ((float(letters) / float(words))))
    # Average Sentences per 100 words
    S = float(100 * ((float(sentences) / float(words))))
    # Using Coleman-Liau index
    index = float(0.0588 * L - 0.296 * S - 15.8)
    # Print appropriate Grade Level
    Count = round(index)
    if (Count < 1):
        print("Before Grade 1")
    elif (Count >= 16):
        print("Grade 16+")
    else:
        print("Grade ", Count)


main()
